app/assets/src/components/ui/controls/dropdowns/BareDropdown.jsx

```diff
@@ -141,17 +158,41 @@ class BareDropdown extends React.Component {
 
     const { filterString } = this.state;
 
+    let wrappedTrigger = trigger;
+
+    // When search is true, semantic-ui doesn't close the dropdown when the trigger is clicked
+    // for some reason. Manually close it in this case.
+    if (this.props.search) {
+      // When search is true, inject an onClick handler onto the trigger element.
+      wrappedTrigger = React.cloneElement(trigger, {
+        onClick: () => {
+          // Manually close the base dropdown if it's open.
+          if (
+            this.props.search &&
+            this.baseDropdownRef.current &&
+            this.baseDropdownRef.current.state.open
```

Comment:
Accessing the state of a child component is generally bad practice. Doing it here only because it's a 3rd party component that doesn't support the behavior we want.

Reply:
Adding this as a comment in the code directly.

Reply:
Isn't this a symptom that we are using the wrong component? Maybe we are officially using dropdown for too much?
It looks to me that we should plan for a long term solution for this...

Reply:
Totally agree. We'd have to do a spike to see how much work replacing the Dropdown component would be. Does this feel high-priority or medium-priority to you?

Reply:
Maybe create a Jira ticket with medium priority for now... and lets chat next week.